app/app.js

```diff
@@ -32,6 +31,118 @@ app.post('/api/buildVisualizationImage/', function (req, res) {
 
 });
 
+app.get('/api/executeImmediate/', (req, res) => {
+    var visualizationid = req.query.visualizationid;
+    jobs.executeImmediate(visualizationid).then(function (result) {
+        res.send(result);
+    }, function (err) {
+        res.send(err);
+    })
+});
+
+app.post('/api/jobSchedule/', function (req, res) {
+    var result = validator.validateReportReqBody(req.body);
+    if (result.error) {
+        logger.log({
+            level: 'error',
+            message: 'error in schedule api due to invalid request body',
+            errMsg: result.error.details[0].message
+        });
+        res.status(422).json({
+            message: result.error.details[0].message.replace(/\"/g, ""),
+        });
+    }
+    else {
+        jobs.createJob(req.body).then(function (result) {
+            if (result.success == 1) {
```

Comment:
You can transform this into one-liner instead of repeating same code structure.
For example:
`res.status(result.success ===1? 201: 302).json({message:result.message})`